### PR TITLE
SGX distribution scripts

### PR DESCRIPTION
**Pull request by amendelzon**
- Moved `dist` to `dist/ledger`
- Added `dist/sgx` with setup and run scripts
- New SGX distribution building script (`build-dist-sgx`)
- Added and updated distribution documentation for both Ledger and SGX
- Additional `HSM2DongleSGX` unit tests for the onboarding operation

### CODE REVIEW

File: dist/sgx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Download and install libssl1.1 from debian main repository
+RUN curl -L -o libssl1.1.deb  https://ftp.debian.org/debian/pool/main/o/openssl/libssl1.1_1.1.1w-0+deb11u1_amd64.deb && \
+    dpkg -i libssl1.1.deb && \
+    rm libssl1.1.deb
+
+# Install SGX runtime dependencies
+RUN echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main' | tee /etc/apt/sources.list.d/intel-sgx.list && \
+    wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - && \
+    apt-get update && \
+    apt-get install libsgx-enclave-common
```
Comment (italo-sampaio) [inline dist/sgx/Dockerfile:17]: Does this work without the `-y` flag?

File: dist/sgx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    apt-get install -y binutils gnupg2 curl wget
+
+# Download and install libssl1.1 from debian main repository
+RUN curl -L -o libssl1.1.deb  https://ftp.debian.org/debian/pool/main/o/openssl/libssl1.1_1.1.1w-0+deb11u1_amd64.deb && \
+    dpkg -i libssl1.1.deb && \
+    rm libssl1.1.deb
+
+# Install SGX runtime dependencies
+RUN echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main' | tee /etc/apt/sources.list.d/intel-sgx.list && \
+    wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - && \
```
Comment (italo-sampaio) [inline dist/sgx/Dockerfile:15]: Minor, but we can remove the dependency on `wget` by using

```
 curl -fsSL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - && \
 ```

**Review by italo-sampaio — APPROVED** [review]
LGTM! Left a few comments in the sgx Dockerfile

File: dist/sgx/Dockerfile (hunk 1 above)
Comment (amendelzon) [reply dist/sgx/Dockerfile]: It seemed to, maybe because the package is small enough? Added the flag just in case.

**Review by italo-sampaio — APPROVED** [review]

**Pull request merged**